Adding type hints and improving docs

dennlinger opened the pull request:
Added type hints to move descriptive comments into actual tangible programming feedback; e.g., it was unclear what type the "document time" should be in. Clarifying that it is a string already helps enormously.  
Also added in comments that should show up during function previews to help clarify what parameters are possible, or what the specific format should be.  
For some parameters, also added direct feedback from Python that raise Errors when inputting incorrect values (if available choices exist in the original Heideltime docs), or help with upper/lowercasing.

I haven't verified whether the hints actually show up (yet), but am planning on at least extending it to a usable state.
@PhilipEHausner: Let me know if you want a more "thorough" PR, happy to add stuff before merging.

dennlinger committed:
Adding type hints and docs.
Added type hints to move descriptive comments into actual tangible programming feedback.
Also added in comments that should show up during function previews to help clarify what parameters are possible.
For some parameters, also added direct feedback from Python that raise Errors when inputting incorrect values (if available choices exist in the original Heideltime docs).

PhilipEHausner commented on python_heideltime/python_heideltime.py:
```diff
@@ -16,76 +16,118 @@
-# documentation: https://gate.ac.uk/gate/plugins/Tagger_GATE-Time/doc/HeidelTime-Standalone-Manual.pdf
+# Taken from the documentation linked below
+AVAILABLE_LANGUAGES = ['ENGLISH', 'GERMAN', 'DUTCH', 'ENGLISHCOLL', 'ENGLISHSCI', 'SPANISH', 'ITALIAN',
+                       'ARABIC', 'VIETNAMESE', 'FRENCH', 'CHINESE', 'RUSSIAN', 'CROATIAN', 'PORTUGUESE', 'ESTONIAN']
```
Personally I would order the languages alphabetically.

PhilipEHausner commented on python_heideltime/python_heideltime.py:
```diff
@@ -16,76 +16,118 @@
+        Only used in combination with 'NEWS' or 'COLLOQUIAL' document types.
+        """
+        if not regex.match(r'[0-9]{4}-[0-9]{2}-[0-9]{2}', document_time):
+            raise ValueError('Incorrect format for document time specified. Please use the \"YYYY-MM-DD\" format.')
```
I think the "s do not need to be escaped here.

PhilipEHausner commented on python_heideltime/python_heideltime.py:
```diff
@@ -16,76 +16,118 @@
+        """
+        if not language.upper() in AVAILABLE_LANGUAGES:
+            raise ValueError(f'Unknown language specified! Please specify one of the supported languages below:\n'
+                             f'{AVAILABLE_LANGUAGES}')
```
Maybe also add what the passed parameter was.

PhilipEHausner commented on python_heideltime/python_heideltime.py:
```diff
@@ -16,76 +16,118 @@
+        """
+        if not doc_type.upper() in AVAILABLE_DOCUMENT_TYPES:
+            raise ValueError(f'Unknown document type specified! Please use one of the following supported document'
+                             f'types: {AVAILABLE_DOCUMENT_TYPES}')
```
same as with language

PhilipEHausner commented on python_heideltime/python_heideltime.py:
```diff
@@ -16,76 +16,118 @@
+        """
+        if not output_type.upper() in AVAILABLE_OUTPUT_TYPES:
+            raise ValueError(f'Unknown output type specified! Plese use one of the following supported output types:'
+                             f'{AVAILABLE_OUTPUT_TYPES}')
```
as above

PhilipEHausner commented on python_heideltime/python_heideltime.py:
```diff
@@ -16,76 +16,118 @@
+                             f'{AVAILABLE_OUTPUT_TYPES}')
         self.output_type = output_type
 
     def set_encoding(self, encoding):
```
type hints

PhilipEHausner commented on python_heideltime/python_heideltime.py:
```diff
@@ -16,76 +16,118 @@
 
-    # True / False
-    def set_verbosity(self, verbosity):
+    # FIXME: Technically, Heideltime supports two different verbosity levels, which we cannot model here.
```
Maybe specify here what you mean with that. I would not remember what I meant here a month alter. ;)

PhilipEHausner reviewed (changes requested):
Thank you for putting up the effort to improve the tool. I would ask you, however, to also add return types to the functions. I know it is always None, but static type checkers expect that if you have type hints you also need a return type (at least with the exception of the constructor I think).

Otherwise I only have minor comments.

dennlinger committed:
Incorporating requested changes:
Adding return type hints;
Adding user inputs in ValueErrors
Sorting potential values alphabetically;
Completing missing type hint for encoding.

dennlinger commented:
Addressed all comments, let me know if you have any further ideas.
As for the explanation in the FIXME, feel free to add further comments if it is still unclear.
